File: hackerrank/problem solving/algorithms/1 - warmup/7 - staircase/staircase.cs
// https://www.hackerrank.com/challenges/staircase/problem?isFullScreen=true

int n = int.Parse(Console.ReadLine()!);
Staircase(n);

// T: O(n^2)
// S: O(n) extra space
void Staircase(int n)
{
    for (int i = 1; i <= n; i++)
    {
        var spaces = new string(' ', n - i);
        var hashes = new string('#', i);
        Console.WriteLine(spaces + hashes);
    }
}

File: hackerrank/problem solving/algorithms/1 - warmup/7 - staircase/staircase.py
# https://www.hackerrank.com/challenges/staircase/problem?isFullScreen=true

def main():
    n = int(input())
    staircase(n)


# T: O(n^2)
# S: O(n) extra space
def staircase(n):
    for i in range(1, n + 1):
        spaces = ' ' * (n - i)
        hashes = '#' * i
        print(spaces + hashes)


if __name__ == '__main__':
    main()
